FileAudioSource/FileSourceAudioDevice.cs

```diff
@@ -0,0 +1,286 @@
+                    capturerSettings = new AudioDeviceSettings();
+                    capturerSettings.NumChannels = wavHeader[22] | (wavHeader[23] << 8);
+                    capturerSettings.SamplingRate = wavHeader[24] | (wavHeader[25] << 8) | (wavHeader[26] << 16) | (wavHeader[27] << 24);
+                    AudioPropertiesChanged?.Invoke(capturerSettings.NumChannels, capturerSettings.SamplingRate);
+                }
+                else
+                {
+                    capturerSettings = new AudioDeviceSettings() { NumChannels = 2, SamplingRate = 48000 };
+                    AudioPropertiesChanged?.Invoke(0, 0);
+                }
+                isAudioCapturerInitialized = true;
+            }
+        }
+
+        public void DestroyAudioCapturer()
+        {
+            lock (audioDeviceLock)
+            {
+                StopAudioCapturer();
+                if (!IsAudioCapturerInitialized())
+                    return;
+                if (sourceFileStream != null)
+                {
+                    sourceFileStream.Close();
+                    sourceFileStream.Dispose();
+                    sourceFileStream = null;
+                }
+                isAudioCapturerInitialized = false;
+            }
+        }
+
+        public void StartAudioCapturer()
+        {
+            lock (audioDeviceLock)
+            {
+                if (IsAudioCapturerStarted())
+                    return;
+                if (!IsAudioCapturerInitialized())
+                    throw new Exception("Audio capturer not initialized");
+
+                isAudioCapturerStarted = true;
+
+                _ = Task.Factory.StartNew(() =>
+                {
+                    int numberOfSamplesPer10ms = capturerSettings.NumChannels * capturerSettings.SamplingRate / 100;
+                    byte[] buffer = new byte[numberOfSamplesPer10ms * 2];
+                    unsafe
+                    {
+                        fixed (byte* bufferPtr = buffer)
+                        {
+                            DateTime nextBatchTime = DateTime.Now;
+                            while (isAudioCapturerStarted)
+                            {
+                                try
+                                {
+                                    if (DateTime.Now >= nextBatchTime)
+                                    {
+                                        nextBatchTime += new TimeSpan(0, 0, 0, 0, 10);
+
+                                        int bytesRead = buffer.Length;
+                                        if (sourceFileStream != null)
+                                            bytesRead = sourceFileStream.Read(buffer, 0, buffer.Length);
+                                        audioBus.WriteCaptureData((IntPtr)bufferPtr, bytesRead / (2 * capturerSettings.NumChannels));
+                                    }
+                                    Thread.Sleep(1);
```

Comment:
Nit. Is there any alternative to the sleep+polling? Perhaps a timer?

Reply:
Mmm... I don't like timers. At least in windows they tend to lag behind. You can never guarantee any kind of precision.
Sleep-polling is the closest I've found to a real-time clock without an active wait that would completely burn the CPU.